src/Cardano/CoinSelection/LargestFirst.hs

```diff
@@ -20,77 +21,140 @@ import Cardano.Types
 import Control.Arrow
     ( left )
 import Control.Monad
-    ( foldM, when )
+    ( foldM )
 import Control.Monad.Trans.Except
     ( ExceptT (..), except, throwE )
 import Data.Functor
     ( ($>) )
 import Data.List.NonEmpty
     ( NonEmpty (..) )
 import Data.Ord
-    ( Down (..), comparing )
+    ( Down (..) )
 
 import qualified Data.List as L
 import qualified Data.List.NonEmpty as NE
 import qualified Data.Map.Strict as Map
 
--- | Implements the __Largest-First__ coin selection algorithm.
+-- | Generate a coin selection according to the __largest first__ algorithm.
+--
+-- For the given /output list/ and /starting UTxO set/, this algorithm
+-- generates a /coin selection/ that is capable of paying for all of the
+-- outputs, and a /remaining UTxO set/ from which spent values have been
```

Comment:
/remaining UTxO set/ is the one that is the result of /starting UTxO set/ - spent values, right?

Reply:
> /remaining UTxO set/ is the one that is the result of /starting UTxO set/ - spent values, right?

Yes, that was my intention.

Do you think there is anything I can add or change to make it clearer?